### ISSUE TITLE
🐛 [Firebase Cloud Messaging] getInitialMessage() called multiple times for a single notification 

### ISSUE THREAD

**Issue by MuhammadFaizan980** [firebase/flutterfire#4188]
## Bug report
Whenever a message is sent from server or any other device, getInitailMessage() gets called 3 times and getInitialMessage stream is being listened multiple times. It is expected to be called only once as notification is processed or read.
I am using **firebase_messaging: ^8.0.0-dev.8**

**Comment by markusaksli-nc** [comment]
Hi @MuhammadFaizan980 
What do you mean? `getInitialMessage` is an asynchronous method that returns a `RemoteMessage` if the app was opened through one. It is not a stream and it is only called by using it in your own code. The method also consumes the initial `RemoteMessage` and any calls to it after the first will return `null`.

This is either a mistake in your code or you are referring to a different method/stream. Please check your code or provide additional clarification as to what you mean.
Thank you

**Comment by MuhammadFaizan980** [comment]
Thanks for your reply @markusaksli-nc, I had a typo (now typo is fixed).
Let me elaborate further.
When **getInitialMessage()** is called or **onMessageOpenedApp ** is listened, it is supposed to consume **RemoteMessage** right? but thing is, it does not, when I open the app from terminated state or suspended state by clicking notification,  respective functions are called but when I come restart app, it again  calls above messages though I have added a null check for **RemoteMessage**, This behavior persists for two to three attempts (that is what I mean by 2-3 times call) before **RemoteMessage** is consumed and **null** is returned.

**Comment by markusaksli-nc** [comment]
Can you please provide a [minimal complete reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example)?

I was not able to reproduce this with the official example using `firebase_messaging: ^8.0.0-dev.8`

<details>
 <summary>flutter doctor -v</summary>

```
[√] Flutter (Channel master, 1.24.0-8.0.pre.365, on Microsoft Windows [Version 10.0.19041.630], locale et-EE)
    • Flutter version 1.24.0-8.0.pre.365 at C:\Development\flutter_master
    • Framework revision 2ceb371d21 (8 hours ago), 2020-11-24 17:18:06 -0800
    • Engine revision 6ed357c8b3
    • Dart version 2.12.0 (build 2.12.0-74.0.dev)

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\marku\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.6953283\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[√] Chrome - develop for the web
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[√] Visual Studio - develop for Windows (Visual Studio Community 2019 16.7.7)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.7.30621.155
    • Windows 10 SDK version 10.0.18362.0

[√] Android Studio (version 4.1.0)
    • Android Studio at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\AndroidStudio\ch-0\201.6953283
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] IntelliJ IDEA Ultimate Edition (version 2020.2)
    • IntelliJ at C:\Users\marku\AppData\Local\JetBrains\Toolbox\apps\IDEA-U\ch-0\202.7660.26
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart

[√] Connected device (5 available)
    • SM G950F (mobile) • ce12171c51cc001c03 • android-arm64  • Android 9 (API 28)
    • Windows (desktop) • windows            • windows-x64    • Microsoft Windows [Version 10.0.19041.630]
    • Web Server (web)  • web-server         • web-javascript • Flutter Tools
    • Chrome (web)      • chrome             • web-javascript • Google Chrome 87.0.4280.66
    • Edge (web)        • edge               • web-javascript • Microsoft Edge 86.0.622.68

• No issues found!
```
</details>

**Comment by MuhammadFaizan980** [comment]
Sorry for late reply, was out of town, here is the minimal code:
`  @override
  void initState() {
    _configureFcm();
    super.initState();
  }

  Future<void> _configureFcm() async {
    _firebaseMessaging = FirebaseMessaging.instance;
    await _firebaseMessaging.requestPermission(
      alert: true,
      announcement: false,
      badge: true,
      carPlay: false,
      criticalAlert: false,
      provisional: false,
      sound: true,
    );

    FirebaseMessaging.onMessage.listen((RemoteMessage message) async {
      // Works fine
    });
    
    FirebaseMessaging.onMessageOpenedApp.listen((RemoteMessage message) { 
      AppToast.showToast(message: 'onMessageOpenedApp called');
      // This gets called three times, every time gets called when app is restarted or I navigate back using pushReplacement()
    });
    
    FirebaseMessaging.instance.getInitialMessage().then((RemoteMessage message){
      AppToast.showToast(message: 'getInitialMessage() called');
      // This gets called 4 times, every time gets called when app is restarted or I navigate back using pushReplacement()
    });
    
  }
`

**Flutter Doctor output:**
`[√] Flutter (Channel stable, 1.22.4, on Microsoft Windows [Version 10.0.10240], locale en-US)
    • Flutter version 1.22.4 at C:\flutter
    • Framework revision 1aafb3a8b9 (13 days ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Program Files\Android\Android Studio1\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[!] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio1
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.50.1)
    • VS Code at C:\Users\My Own\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.15.1

[√] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

! Doctor found issues in 1 category.
`

**Comment by MuhammadFaizan980** [comment]
<details>
<summary> flutter doctor -v output</summary>
<br>
[√] Flutter (Channel stable, 1.22.4, on Microsoft Windows [Version 10.0.10240], locale en-US)
    • Flutter version 1.22.4 at C:\flutter
    • Framework revision 1aafb3a8b9 (13 days ago), 2020-11-13 09:59:28 -0800
    • Engine revision 2c956a31c0
    • Dart version 2.10.4

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Program Files\Android\Android Studio1\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[!] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio1
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.50.1)
    • VS Code at C:\Users\My Own\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.15.1

[√] Connected device (1 available)
    • Android SDK built for x86 (mobile) • emulator-5554 • android-x86 • Android 10 (API 29) (emulator)

! Doctor found issues in 1 category.

</details>

**Comment by markusaksli-nc** [comment]
Do you mean just hot restarting the app or actually terminating and re-opening it?

**Comment by MuhammadFaizan980** [comment]
I tested on hot restart and full app restart both

**Comment by MuhammadFaizan980** [comment]
Issue persists not only with restarting of the app but also with the replacement of screen using pushReplacement()

**Comment by markusaksli-nc** [comment]
Do you experience this with the code in the [official example](https://github.com/FirebaseExtended/flutterfire/tree/master/packages/firebase_messaging/firebase_messaging/example/lib) as well?

Though this shouldn't be possible in the first place.

**Comment by MuhammadFaizan980** [comment]
I followed flutter fire documentation but will try pfficial example as well let you know

**Comment by savanbthakkar** [comment]
We used the same example on this page: https://pub.dev/packages/firebase_dynamic_links/example
And we still have the same problem. We have this code on our home page in our app. Then based on the input, we redirect the user to another page. User can then go back to previous page. In this case, it gets data to PendingDynamicLinkData variable from FirebaseDynamicLinks.instance.getInitialLink() again and redirects the user back to second page. Hence causing infinite loop.
I have a workaround, to save the incoming URLs in a temporary variable, but that's a dirty workaround.
Please suggest if there is a better way to solve this.

**Comment by MuhammadFaizan980** [comment]
Sorry, was out of country, as for the official example, I tried it as well but no luck, same issue.

> Hey @MuhammadFaizan980. We need more information to resolve this issue but there hasn't been an update in 7 weekdays. I'm marking the issue as stale and if there are no new updates in the next 7 days I will close it automatically.
> 
> If you have more information that will help us get to the bottom of this, just add a comment!

**Comment by MuhammadFaizan980** [comment]
Ok, here is the new finding after upgrading to x.11 version.
Now the only issue of repeated recalling of getInitialMessage() remains, as for the onMessageOpenedApp stream, it is working fine now, it sends message value just once, after that, RemoteMessage is null.

**Comment by Fabio-ciao** [comment]
Hi @MuhammadFaizan980 Did you figure it out?
I have your same problem here with **firebase_messaging: 9.1.1**

<img width="600" alt="Schermata 2021-04-15 alle 11 33 21" src="https://user-images.githubusercontent.com/53313420/114847814-6784f780-9dde-11eb-8599-5bbb5acab797.png">

**Comment by MuhammadFaizan980** [comment]
> Hi @MuhammadFaizan980 Did you figure it out?
> I have your same problem here with **firebase_messaging: 9.1.1**
> 
> <img alt="Schermata 2021-04-15 alle 11 33 21" width="600" src="https://user-images.githubusercontent.com/53313420/114847814-6784f780-9dde-11eb-8599-5bbb5acab797.png">

I solved the issue by saving notification ID into shared preferences and checking every time if the id already exists or not, if exists, I do not take any action in the listeners.
Following is my code for handling listener: 
`  void _handleNotificationsListeners() {
    FirebaseMessaging.onMessageOpenedApp.listen((RemoteMessage message) {
      // works fine |Y|
      if (message != null) Utils.showToast('onMessageOpenedApp');
    });
    FirebaseMessaging.instance
        .getInitialMessage()
        .then((RemoteMessage message) async {
      // fixed, working with a minor tweak |Y|
      if (message != null &&
          (await Utils.getFcmId(message.messageId) == null)) {
        Utils.showToast('getInitialMessage');
        Utils.setFcmId(message.messageId);
      }
    });
    FirebaseMessaging.onMessage.listen((RemoteMessage message) {
      // works fine |Y|
      if (message != null) Utils.showToast('onMessage');
    });
  }`

 `Utils.setFcmId()` is as following:
`  static Future<void> setFcmId(String fcmId) async {
    var pref = await SharedPreferences.getInstance();
    await pref.setString(fcmId, fcmId);
  }`

 `Utils.getFcmId()` is as following:

` static Future<String> getFcmId(String fcmId) async {
    return (await SharedPreferences.getInstance()).getString(fcmId) ?? null;
  }`

**Comment by MuhammadFaizan980** [comment]
> Hi @MuhammadFaizan980 Did you figure it out?
> I have your same problem here with **firebase_messaging: 9.1.1**
> 
> <img alt="Schermata 2021-04-15 alle 11 33 21" width="600" src="https://user-images.githubusercontent.com/53313420/114847814-6784f780-9dde-11eb-8599-5bbb5acab797.png">

Yes I did, I used shared preferences in combination with the FCM plugin and it works fine. I store FCM IDs into shared preferences and check every time if the notiication ID already exists in the prefs or not, if it does, do not take any action.
Following is the code:
    

    void _handleNotificationsListeners() {
    FirebaseMessaging.onMessageOpenedApp.listen((RemoteMessage message) {
      // works fine |Y|
      if (message != null) Utils.showToast('onMessageOpenedApp');
    });
    FirebaseMessaging.instance
        .getInitialMessage()
        .then((RemoteMessage message) async {
      // fixed, working with a minor tweak |Y|
      if (message != null &&
          (await Utils.getFcmId(message.messageId) == null)) {
        Utils.showToast('getInitialMessage');
        Utils.setFcmId(message.messageId);
      }
    });
    FirebaseMessaging.onMessage.listen((RemoteMessage message) {
      // works fine |Y|
      if (message != null) Utils.showToast('onMessage');
    });
    }

And following functions are used to set and get FCM IDs:

     static Future<void> setFcmId(String fcmId) async {
      var pref = await SharedPreferences.getInstance();
      await pref.setString(fcmId, fcmId);
    }

    static Future<String> getFcmId(String fcmId) async {
      return (await SharedPreferences.getInstance()).getString(fcmId) ?? null;
    }

**Comment by Fabio-ciao** [comment]
Thank you very much !!

**Comment by IlyaBlinov** [comment]
Hi. Method getInitialMessage() return null and FirebaseMessaging.onMessageOpenedApp not working when I clicked on push notification. How to fix it? Do you have any ideas to fix it?

**Comment by Fabio-ciao** [comment]
> Hi. Method getInitialMessage() return null and FirebaseMessaging.onMessageOpenedApp not working when I clicked on push notification. How to fix it? Do you have any ideas to fix it?

Hi, take a look at the official example https://pub.dev/packages/firebase_messaging/example
For me the only problem was the .getInitialMessage() being called multiple times, but you can solve this using Shared Preferences as shown above

**Comment by MuhammadFaizan980** [comment]
> Hi. Method getInitialMessage() return null and FirebaseMessaging.onMessageOpenedApp not working when I clicked on push notification. How to fix it? Do you have any ideas to fix it?

If you are not receiving any notification at all, even if the app is in the background, add the firebase messagin dependency in the app level build.gradle file and it will work fine

**Comment by BhavyKoshti9spl** [comment]
i am not getting data while opening app from the notification click RemoteMessage is giving null.

**Comment by TusharRoy23** [comment]
> > Hi @MuhammadFaizan980 Did you figure it out?
> > I have your same problem here with **firebase_messaging: 9.1.1**
> > <img alt="Schermata 2021-04-15 alle 11 33 21" width="600" src="https://user-images.githubusercontent.com/53313420/114847814-6784f780-9dde-11eb-8599-5bbb5acab797.png">
> 
> Yes I did, I used shared preferences in combination with the FCM plugin and it works fine. I store FCM IDs into shared preferences and check every time if the notiication ID already exists in the prefs or not, if it does, do not take any action.
> Following is the code:
> 
> ```
> void _handleNotificationsListeners() {
> FirebaseMessaging.onMessageOpenedApp.listen((RemoteMessage message) {
>   // works fine |Y|
>   if (message != null) Utils.showToast('onMessageOpenedApp');
> });
> FirebaseMessaging.instance
>     .getInitialMessage()
>     .then((RemoteMessage message) async {
>   // fixed, working with a minor tweak |Y|
>   if (message != null &&
>       (await Utils.getFcmId(message.messageId) == null)) {
>     Utils.showToast('getInitialMessage');
>     Utils.setFcmId(message.messageId);
>   }
> });
> FirebaseMessaging.onMessage.listen((RemoteMessage message) {
>   // works fine |Y|
>   if (message != null) Utils.showToast('onMessage');
> });
> }
> ```
> 
> And following functions are used to set and get FCM IDs:
> 
> ```
>  static Future<void> setFcmId(String fcmId) async {
>   var pref = await SharedPreferences.getInstance();
>   await pref.setString(fcmId, fcmId);
> }
> 
> static Future<String> getFcmId(String fcmId) async {
>   return (await SharedPreferences.getInstance()).getString(fcmId) ?? null;
> }
> ```

I got your solution. But what if a user didn't clear that fcmId Shared preference. Is he able to go to his target page again? 
Just asking...

**Comment by ParasGarg7** [comment]
I observed the same thing, the **RemoteMessage** wasn't consumed in the **onMessageOpenedApp()**.([doc](https://firebase.flutter.dev/docs/messaging/notifications#handling-interaction))
My use case is that I am calling the below mentioned function on landing on the Dashboard Screen, which is the first Screen the user interacts with, and then on the notification selection, I am resetting the user to the Dashboard Screen irrespective of which screen the user is upon and then redirecting him from there. This reset again calls the below mentioned func, and thus the **getInitialMessage()** is called 2-3 times, before it is actually consumed, causing a loop.

> Thanks for your reply @markusaksli-nc, I had a typo (now typo is fixed).
> Let me elaborate further.
> When **getInitialMessage()** is called or **onMessageOpenedApp** is listened, it is supposed to consume **RemoteMessage** right? but thing is, it does not, when I open the app from terminated state or suspended state by clicking notification, respective functions are called but when I restart app, it again calls above functions with **RemoteMessage** still not null as I have added a null check for **RemoteMessage**, This behavior persists for two to three attempts (that is what I mean by 2-3 times call) before **RemoteMessage** is consumed and **null** is returned.

![Screenshot 2021-09-03 at 10 57 35 AM](https://user-images.githubusercontent.com/31133978/131955007-d7dbac68-ab56-4118-995a-bf6893134cda.png)

**Comment by anujsaxenadev** [comment]
> I observed the same thing, the **RemoteMessage** wasn't consumed in the **onMessageOpenedApp()**.([doc](https://firebase.flutter.dev/docs/messaging/notifications#handling-interaction))
> My use case is that I am calling the below mentioned function on landing on the Dashboard Screen, which is the first Screen the user interacts with, and then on the notification selection, I am resetting the user to the Dashboard Screen irrespective of which screen the user is upon and then redirecting him from there. This reset again calls the below mentioned func, and thus the **getInitialMessage()** is called 2-3 times, before it is actually consumed, causing a loop.
> 
> > Thanks for your reply @markusaksli-nc, I had a typo (now typo is fixed).
> > Let me elaborate further.
> > When **getInitialMessage()** is called or **onMessageOpenedApp** is listened, it is supposed to consume **RemoteMessage** right? but thing is, it does not, when I open the app from terminated state or suspended state by clicking notification, respective functions are called but when I restart app, it again calls above functions with **RemoteMessage** still not null as I have added a null check for **RemoteMessage**, This behavior persists for two to three attempts (that is what I mean by 2-3 times call) before **RemoteMessage** is consumed and **null** is returned.
> 
> ![Screenshot 2021-09-03 at 10 57 35 AM](https://user-images.githubusercontent.com/31133978/131955007-d7dbac68-ab56-4118-995a-bf6893134cda.png)

I have your same problem with firebase_messaging: ^10.0.6
As it is stated in documentation -
https://firebase.flutter.dev/docs/messaging/notifications/#handling-interaction

That "**Once consumed, the RemoteMessage will be removed**." but it is not happening.
**getInitialMessage** retuning **RemoteMessage** after it is used.

I can have a workaround but this is just dirty code -

```
class FirebaseManager {
  bool consumedOnce = false;

  getInitialMessageFromMessenging() async {
    if (!consumedOnce) {
      RemoteMessage? initialMessage =
          await FirebaseMessaging.instance.getInitialMessage();
      if (initialMessage != null) {
        _onSelectNotification(jsonEncode(initialMessage.data));
      }
      consumedOnce = true;
    }
  }
}
```

**Comment by russellwheatley** [comment]
@MuhammadFaizan980, I've just tested your code sample in the official example and `getInitialMessage`  only fire once. Do you have a complete example? 

Also - there is a [PR for another messaging issue](https://github.com/FirebaseExtended/flutterfire/pull/6961) where the `FirebaseMessaging` instance wasn't being cached so I wonder if this is another bug as a result of that.

**Comment by MuhammadFaizan980** [comment]
@russellwheatley I used the official example. Seems like this issue related to [https://github.com/FirebaseExtended/flutterfire/pull/6961#issue-990974823](url)

**Comment by vytautas-pranskunas-** [comment]
watching

**Comment by russellwheatley** [comment]
@MuhammadFaizan980 The latest `firebase_messaging` (`10.0.7`) has been released with that fix. Let me know if that resolves your issue 😄.

**Comment by MuhammadFaizan980** [comment]
@russellwheatley I tested the latest version but still error persists. I sent the notification when app was in the background, `onMessageOpenedApp()` was called, which is normal but then I restarted the app without sending any new notification and `getInitialMessage()` was called which is **NOT NORMAL** behavior..
In second scenario, I sent a notification when app was closed, tapped the notification and `getInitialMessage()` was called which is normal behavior, I then restarted the app without sending any new notification and `getInitialMessage()` was called again which is **NOT NORMAL** behavior.

**Comment by joeloudjinz** [comment]
Facing same problem here firebase messaging `10.0.7` and flutter `2.5.0`. followed fireflutter docs to use FCM

**Comment by daniloapr** [comment]
> Facing same problem here firebase messaging `10.0.7` and flutter `2.5.0`. followed fireflutter docs to use FCM

Same here, with flutter `2.5.1` and firebase_messaging `10.0.7`.

**Comment by Alecinvader** [comment]
> 
> 
> Facing same problem here firebase messaging `10.0.7` and flutter `2.5.0`. followed fireflutter docs to use FCM

Also having same issue with most up to date issue, the `RemoteMessage? initialMessage = await FirebaseMessaging.instance.getInitialMessage();` is not getting consumed on first call, which if you re-init that page like a "home.dart" it will keep triggering that notification.

**Comment by flutter-clutter** [comment]
Still experiencing this issue with firebase_messaging 10.0.9

**Comment by abit-t-vlad** [comment]
Still experiencing this issue with version 11.1.0.

**Comment by khannah** [comment]
Same issue here on iOS, firebase_messaging: ^11.1.0

**Comment by khang-dev** [comment]
Same issue here on iOS, firebase_messaging: ^11.1.0, flutter 2.5.3

**Comment by lamobratory** [comment]
Same here on firebase_messaging: ^11.2.0

Trying a workaround to cache it locally as the instance lives with the app, no need for user preferences:

```
  String? _lastConsummedMessageId;

  @override
  Future<PushNotification?> checkAppOpenedFromNotifications() async {
    PushNotification? pn;
    final message = await FirebaseMessaging.instance.getInitialMessage();

    if (_messageAlreadyConsumed(message?.messageId)) return null;

    if (message != null) {
      pn = PushNotification.fromRemoteMessage(message);
    }

    return pn;
  }

  @override
  void tappedNotificationHandler(
      Function(PushNotification) tappedNotificationCallback) {
    FirebaseMessaging.onMessageOpenedApp.listen((RemoteMessage message) {
      if (_messageAlreadyConsumed(message.messageId)) return;

      final pn = PushNotification.fromRemoteMessage(message);
      tappedNotificationCallback(pn);
    });
  }

  @override
  void listenToForegroundMessages(String notificationIcon,
      Function(PushNotification) tappedNotificationCallback) {
    FirebaseMessaging.onMessage.listen((RemoteMessage message) {
      if (_messageAlreadyConsumed(message.messageId)) return;

      _createLocalNotification(
          message, notificationIcon, tappedNotificationCallback);
    });
  }

  bool _messageAlreadyConsumed(String? newMessageId) {
    if (_lastConsummedMessageId == newMessageId) {
      return true;
    }
    _lastConsummedMessageId = newMessageId;
    return false;
  }
```

**Comment by otopba** [comment]
Still experiencing this issue with firebase_messaging 11.2.0

**Comment by duzenko** [comment]
One year and three major versions and later, still no action on this basic functionality failure
You'd think they could be at least bothered to remove 
```
  /// Once the [RemoteMesage] has been consumed, it will be removed and further
  /// calls to [getInitialMessage] will be `null`.
```
from the autodoc, huh?

**Comment by darshankawar** [comment]
Everyone in this thread facing this behavior, please upvote the original issue instead of commenting `facing same issue`, the upvote will help the team to look into this on priority. Since this is already labeled, I'll try to forward this to the team for their look-in.

/cc @russellwheatley for further insights on this.

**Comment by vytautas-pranskunas-** [comment]
Any progress?

**Comment by sherkot** [comment]
Any news?

**Comment by brunovsiqueira** [comment]
Any updates?

**Comment by MuhammadFaizan980** [comment]
@brunovsiqueira issue is still there, please re-open this issue.

**Comment by LeGoffMael** [comment]
@Salakar this issue is still reproductible (in release mode only in my case cf #7955) using`firebase_messaging: ^11.2.10`.
I don't fix this is fixed yet.

**Comment by darshankawar** [comment]
Did you try using the latest plugin version https://pub.dev/packages/firebase_messaging/changelog#11210 ?

**Comment by MuhammadFaizan980** [comment]
> Did you try using the latest plugin version https://pub.dev/packages/firebase_messaging/changelog#11210 ?

@LeGoffMael mentioned that he was using latest version i.e `11.2.10`

**Comment by darshankawar** [comment]
Sorry, I missed that.

/cc @Salakar

**Comment by MuhammadFaizan980** [comment]
@russellwheatley same here, not sure why this issue is still there, I used latest version and still the same issue, even using the official example code has this issue

**Comment by justjew** [comment]
Same on `firebase_messaging: ^11.4.0`

**Comment by kaleemAsadEpyphite** [comment]
same on   firebase_messaging: ^11.2.14

**Comment by VovaMedvediev** [comment]
Actually, it isn't fixed.
getInitialMessage() is called 2-3 times on Android.

**Comment by MuhammadFaizan980** [comment]
exactly @VovaMedvediev, I don;t know why this was marked as closed while the issue still persists

**Comment by darshankawar** [comment]
@MuhammadFaizan980 
Can you provide logs where it shows the issue still persists along with ? Are you seeing this on Android or iOS ?

The fix made by the team which closed this issue, was for iOS mainly if you look at the PR: https://github.com/firebase/flutterfire/pull/9315

If you are seeing this issue on Android specifically, please file a new issue for better tracking.
